sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -214,10 +218,15 @@ private[sql] class JacksonGenerator(
   private def writeMapData(
       map: MapData, mapType: MapType, fieldWriter: ValueWriter): Unit = {
     val keyArray = map.keyArray()
+    val keyString = mapType.keyType match {
+      case CalendarIntervalType =>
+        (i: Int) => IntervalUtils.toMultiUnitsString(keyArray.getInterval(i))
+      case _ => (i: Int) => keyArray.get(i, mapType.keyType).toString
```

Comment:
it's fragile to rely on `toString`. e.g. `UnsafeRow.toString` is not human readable. Shall we recursively write map key as json object? cc @HyukjinKwon

Reply:
Ah, I am sorry I missed this cc. in JSON the key should be a string. We should either make it string always or explicitly disallow.

Reply:
cc @viirya I think we talked about this before.

Reply:
Yea, I think currently the map key is not very useful for some types. To make human readable map keys, we need do specific serialization for some map key types. Maybe I create a JIRA ticket to follow it up?

Reply:
Yeah .. +1 !

Reply:
Created https://issues.apache.org/jira/browse/SPARK-29946 to follow it up.